src/crystal/system/win32/file_descriptor.cr

```diff
@@ -275,46 +275,58 @@ module Crystal::System::FileDescriptor
     end
     ret
   end
+end
 
-  private def console_mode?(handle)
-    LibC.GetConsoleMode(handle, out _) != 0
-  end
+private class ConsoleUtils
```

Comment:
Not sure if we even need a separate namespace for this. But it should rather be `module` than `class`.
I think we could just put everything directly into `Crystal::System::FileDescriptor` though.

Reply:
You're right. I didn't know before that class variables could be defined in the module.

I prefer a separate private namespace. If there is a better implementation in the future, it will be easy to modify.